static.md

```diff
@@ -0,0 +1,36 @@
+# Statics
+
+Statics (`static`, `static mut`) are the simplest kind of compile-time evaluated data:
+* The user explicitly requested them to be evaluated at compile-time,
+  so evaluation errors from computing the initial value of a static are no concern
+  (in other words, [const safety](const_safety.md) is mostly not an issue).
+* They observably get evaluated *once*, with the result being put at some address known at run-time,
+  so there are no fundamental restrictions on what statics can do.
+* The compiler checks that statics are `Sync`, justifying sharing their address across threads.
+* [Constants](const.md) and [promoteds](promotion.md) are not allowed to read from statics,
+  so their final value does not have have to be [const-valid](const_safety.md#const-safety-check-on-values) in any meaningful way.
+  As of 2019-08, we do check them for validity anyway, to be conservative; and indeed constants could be allowed to read from frozen statics.
```

Comment:
I think this would be nice personally; less times to invent an ephemeral constant just to please the check and you also get to use exported statics from other crates.

Any future compat hazards with some new feature?

Reply:
Discovered a problem... If we want to have a scheme like https://github.com/rust-lang/rfcs/pull/2492 but centered around `static`s instead, allowing `static`s in const contexts will give us all the problems re. separate compilation.

Reply:
@Centril sorry, I do not follow... why is it harder for separate compilation to support `const` referring to `static` than `const` referring to `const` (and the latter we already support)?

Reply:
In this case it would be a `static` for which the value isn't known yet; that is something like `pub extern static Foo: u8;` <-- no value. That would be similar to https://github.com/rust-lang/rfcs/pull/2492 in that it is matched up with something in the crate graph. If you can then refer to `Foo` in types then that will create all the same problems as with `extern existential type`.

Reply:
As long as whatever *instantiates* the `static` checks them for validity, we are sill good. The important part is the invariant that all statics have been checked. The order doesn't matter.